src/main/java/com/github/cameltooling/lsp/internal/completion/modeline/CamelKModelineInsertionProcessor.java

```diff
@@ -0,0 +1,60 @@
+package com.github.cameltooling.lsp.internal.completion.modeline;
+
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.TextDocumentItem;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+public class CamelKModelineInsertionProcessor {
+
+    private final TextDocumentItem textDocumentItem;
+
+    public CamelKModelineInsertionProcessor(TextDocumentItem textDocumentItem) {
+        this.textDocumentItem = textDocumentItem;
+    }
+
+    public CompletableFuture<List<CompletionItem>> getInsertion() {
+        return CompletableFuture.completedFuture(
+                Arrays.asList(
+                        getCompletionCorrespondingToDocument()
+                ));
+    }
+
+    private CompletionItem getCompletionCorrespondingToDocument() {
+        //What to throw here?
```

Comment:
Can you explain what will be the purpose of throwing an Exception?

Reply:
My first impression woudl be more to write a test for these cases and ensure that it doesn't explode. Like for instance no completion is returned and that's it.

Reply:
I just did that. Although the unit tests are more like integration tests but those classes are built to work together so I didn't see the value on making unit tests for each of them.